File: ServiceBook/ClassLibrary1/Models/Service.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ServiceBook.DAL.Models
{
    public class Service
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int ID { get; set; }
        public DateTime ServiceDate { get; set; }
        public int Flag { get; set; }
        public DateTime NextVisitDate{ get; set; }
        public int CurrentKm { get; set; }
        public int NextVisitKm { get; set; }
        public double Price { get; set; }
        public int VID { get; set; }
        public int WPID { get; set; }
        public int? RID { get; set; }
        

        [ForeignKey("VID")]
        public virtual Vehicle Vehicle { get; set; }
        [ForeignKey("WPID")]
        public virtual WorkingPoint WorkingPoint{ get ;set; }

        [ForeignKey("RID")]
        public virtual Review Review { get; set; }
        public virtual ICollection<SSI> SSI { get; set; }
        public virtual ICollection<SE> SE { get; set; }
       


    }
}

File: ServiceBook/ServiceBook/Scripts/Service/Models/Service.js
function Service(data)
{
    
    this.id=null;
    this.serviceDate=null;
    this.flag=null;
    this.nextVisitDate=ko.observable(null);
    this.currentKm=ko.observable(0);
    this.nextVisitKm=ko.observable(0);
    this.vehicle = null;
    this.serviceInterventions = ko.observableArray(null);
    this.employees = ko.observableArray(null);
    this.price = ko.observable(0);
    this.workingPoint=ko.observable("");
    this.companyName=ko.observable("");
    if(data!=null)
    {
        this.id=data.ID;
        this.serviceDate=convertToFormatJs(data.ServiceDate);
        this.flag=data.Flag;
        this.nextVisitDate(convertToFormatJs(data.NextVisitDate));
        this.currentKm(data.CurrentKm);
        this.workingPoint(data.WorkingPoint);
        this.companyName(data.CompanyName);
        this.nextVisitKm(data.NextVisitKm);
        this.vehicle = new Vehicle(data.Vehicle);
        this.price(data.Price);
        var serviceInterventions = _.map(data.ServiceInterventions, function (serviceInterventions, index) {
            return new ServiceIntervention(serviceInterventions);

        });
        this.serviceInterventions(serviceInterventions);
      

        var employees = _.map(data.Employees, function (employee, index) {
            return new Employee(employee);

        });
        this.employees(employees);
    }
}


